application/frontend/controllers/MfaController.php

```diff
@@ -179,12 +188,28 @@ protected function getRequestedMfa($id)
             );
         }
 
+        return $mfa;
+    }
+
+    /**
+     * Find an MFA by id and employee_id and call its loadData method
+     *
+     * @param int $id
+     * @return Mfa|null
+     * @throws BadRequestHttpException
+     * @throws NotFoundHttpException
+     */
+    protected function getRequestedMfa($id): Mfa
+    {
+        $mfa = $this->getAuthorizedMfa($id);
+
         // rpOrigin is needed for WebAuthn authentication
         $rpOrigin = \Yii::$app->request->get('rpOrigin', '');
         if ($rpOrigin != '' && !in_array($rpOrigin, \Yii::$app->params['authorizedRPOrigins'])){
             throw new ForbiddenHttpException("Invalid rpOrigin", 1638539680);
         }
         $mfa->loadData($rpOrigin);
+        $mfa->loadMfaWebauthns();
```

Comment:
What would you think of moving the logic currently in `loadMfaWebauthns()` into the `loadData()` function? It seems like it's also loading extra information into the `data` field, just for a specific type of MFA, and `loadData()` already has some conditions related to loading different data for specific types of MFAs.

Reply:
`loadData` is also called by User::getVerifiedMfaOptions(), which doesn't call `loadMfaWebauthns`.
I don't know if it matters. Knowing that, do you have a preference?

Reply:
Yeah, it's fine for that to include the MFA WebAuthn data as well. (So merge it in with `loadData()`).